internal/rukpak/source/containers_image.go

```diff
@@ -171,6 +195,35 @@ func (i *ContainersImageRegistry) Cleanup(_ context.Context, bundle *BundleSourc
 	return deleteRecursive(i.bundlePath(bundle.Name))
 }
 
+func (i *ContainersImageRegistry) checkUpdatedConfiguration() error {
+	i.checkConfigLock.Lock()
+	defer i.checkConfigLock.Unlock()
+	dirEntries, err := os.ReadDir(containersConfigFolder)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		// Ignore not found errors
+		return fmt.Errorf("could not read registries configuration directory, error: %w", err)
+	}
+	for _, entry := range dirEntries {
+		if entry.Type().IsRegular() || !strings.HasPrefix(entry.Name(), "..") {
+			// Skip 'normal' entries
+			continue
+		}
+		// Trim the leading '..' for symlinks
+		trimmed := strings.TrimPrefix(entry.Name(), "..")
+		t, err := time.Parse(mountedSymlinkTimestampFormat, trimmed)
+		if err != nil {
+			continue
+		} else if !t.Equal(i.RegistriesConfTimestamp) {
+			// We've found the timestamped symlink and it's been updated
```

Comment:
@dtfranz @joelanford 

turns out there might be an issue with relying on created symlinks: if you use `subPath` to mount a specific file from ConfigMap, the symlinks are not created at all. See https://stackoverflow.com/a/50687707 and https://github.com/kubernetes/kubernetes/issues/50345 for context.

Note: _can't seem to have the powers to unresolve a previous discussion, so adding it here._

Reply:
Given that, an `fsnotify` solution is starting to seem more attractive to me now :sweat_smile:

Reply:
Going back to a question I saw @perdasilva ask.. What is the cost of invalidating every single time? I assume that would mean that `Unpack` would be re-reading files and re-loading essentially the entire configuration at every pass.

WDYT about starting out with "invalidate every time" to keep code complexity low, with a comment that explains that we pay this reload cost unnecessarily, and that if it becomes a performance issue, we should follow up later and optimize to invalidate only on changes.

Reply:
I'm fine with that personally. Have we encountered any performance issues otherwise so far? Just so we aren't adding on top of anything else. AFAIK we aren't, so I'd be OK with invalidating on each pass with a follow up issue to track it.

Reply:
@perdasilva now kicking myself because I completely misinterpreted that question you asked - my bad!! I thought you were saying my first change was invalidating on each pass and didn't read it as a recommendation :sweat_smile: :sweat_smile:

Reply:
Running an extremely rudimentary time elapsed check on a local e2e run, the cache invalidation can take between 500ns and 3µs. I think that's acceptable though it may be worth mentioning every unpack will have to wait on the config mutex from any prior unpack.

Reply:
all good ^^